src/averageAges.js

```diff
@@ -55,7 +70,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person => onlyWithSon
+    ? person.sex === 'f' && people.some(children =>
+      (person.name === children.mother) && children.sex === 'm')
+    : person.sex === 'f' && people.some(children =>
+      person.name === children.mother));
+
+  const child = people.filter(person => onlyWithSon
+    ? person.sex === 'm' && people.some(mother =>
+      (mother.name === person.mother))
+    : people.some(mother =>
+      (mother.name === person.mother)));
+
+  const averageAgeDifference = child.map(person => (
+    person.born - mothers.find(mother =>
```

Comment:
you can use here people and don't find mothers

```suggestion
    person.born - people.find(mother =>
```